[13.0] sale_by_packaging: no pkging override if already set

simahawk opened the pull request:
Before this change the packaging was always overridden
hence the user had not control on the packaging to be used on SO line.

TODO:

- [x] rewrite tests

simahawk committed:
sale_by_packaging: no pkging override if already set

Before this change the packaging was always overridden
hence the user had not control on the packaging to be used on SO line.

simahawk committed:
sale_by_packaging: skip auto assign when forcing uom qty

When the product_uom_qty is computed from inverse method of packaging qty
the packaging is already there and there's no need to auto-assign.

This also prevents skipping direct writes on prod uom qty or other keys
other than product_packaging.

TDu commented on sale_by_packaging/models/sale_order_line.py:
```diff
@@ -101,14 +101,22 @@ def _get_product_packaging_having_multiple_qty(self, product, qty, uom):
         for line in self:
+            if line.product_packaging:
+                # don't touch it if set already
+                continue
```
This is problematic in my view, I open a quotation change the quantity on a line, click save, there is no message and the changes are not saved

sebalix replied:
fixed

jgrandguillaume reviewed (approved):
LGTM

TDu commented:
@simahawk  Looks good as it is now thanks to @sebalix for the last fix. Does it still need the `need fixing` and `work in progress` labels ?

simahawk committed:
sale_by_packaging: drop useless check on pkging multiple qty

This check is already done by core sale module.
Tests where not checkingt for that specific exception.

simahawk committed:
sale_by_packaging: fix prod change override packaging

Packaging and qty should be overridden only if not set yet

simahawk committed:
sale_by_packaging: fix tests

simahawk commented:
fixes integrated into https://github.com/OCA/sale-workflow/pull/1276
